From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect struct musb_platform_ops
Patch-mainline: never, kabi
References: kabi

In 4.4.42, commit 2a0e2a89cc414349d259e7e06187b467abb07596 (usb: musb:
core: add clear_ep_rxintr() to musb_platform_ops), upstream commit
6def85a396ce7796bd9f4561c6ae8138833f7a52 added clear_ep_rxintr to
struct musb_platform_ops which caused kABI checker to complain.

The structure is internal-only, so protect the change by __GENKSYMS__
and be done with it.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 drivers/usb/musb/musb_core.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/usb/musb/musb_core.h
+++ b/drivers/usb/musb/musb_core.h
@@ -214,7 +214,9 @@ struct musb_platform_ops {
 				dma_addr_t *dma_addr, u32 *len);
 	void	(*pre_root_reset_end)(struct musb *musb);
 	void	(*post_root_reset_end)(struct musb *musb);
+#ifndef __GENKSYMS__
 	void	(*clear_ep_rxintr)(struct musb *musb, int epnum);
+#endif
 };
 
 /*
